src/modules/users/controllers/users.js

```diff
@@ -0,0 +1,86 @@
+const bcrypt = require('bcrypt')
+
+const User = require('../models/User')
+const usernameGenerator = require('../utils/usernameGenerator')
+
+const userModel = User()
+
+const create = async (req, res) => {
+  const { nome, senha, idPapel } = req.body
+
+  try {
+    const username = usernameGenerator.createUsernameFromName(nome)
+
+    const hashPassword = await bcrypt.hash(senha, 10)
+
+    const userId = await userModel.create({ nome: username, senha: hashPassword, idPapel })
+
+    res.status(200).send({ id: userId, nome: username })
+  } catch (error) {
+    res.status(400).json({ error: error.message })
+  }
+}
+
+const list = async (req, res) => {
+  try {
+    const events = await userModel.list()
+
+    res.send(events)
+  } catch (error) {
+    res.status(400).json({ error: error.message })
```

Comment:
Por que Bad Request (400)? A API não está usando nenhum valor informado por um usuário, não é possível ela quebrar por mal uso.

Reply:
Talvez seja o caso de retornar 500.

Reply:
Vou ver se não é por causa de autenticação, se não for, mudo pra 500